build/targets/common.props

```diff
@@ -1,6 +1,7 @@
 <Project>
   <PropertyGroup>
-    <BuildDir>$(RepositoryRoot)\build</BuildDir>
-    <ArtifactsDir>$(BuildDir)\artifacts</ArtifactsDir>
+    <RepositoryRoot Condition="'$(RepositoryRoot)'==''">$(MSBuildThisFileDirectory)..\..\</RepositoryRoot>
```

Comment:
MSBuildThisFileDirectory refers to the location of .build/targets/common.props, not the location of test projects. This value isn't affect by nested test directories.